actions/drain_node_handler.go

```diff
@@ -202,10 +237,20 @@ func (h *drainNodeHandler) listNodePodsToEvict(ctx context.Context, node *v1.Nod
 		return nil, fmt.Errorf("listing node %v pods: %w", node.Name, err)
 	}
 
-	podsToEvict := lo.Filter(pods.Items, func(pod v1.Pod, _ int) bool {
-		return !isDaemonSetPod(&pod) && !isStaticPod(&pod)
-	})
+	podsToEvict := make([]v1.Pod, 0)
+	castPods := make([]v1.Pod, 0)
+	// Evict CAST PODs as last ones
+	for _, p := range pods.Items {
+		if p.Namespace == h.cfg.castNamespace && !isDaemonSetPod(&p) && !isStaticPod(&p) {
```

Comment:
Wouldn't be also interesting filter the pods that are possible to evict?

Reply:
Yes, good idea. I would add filter for pod that are already in deleting state/phase
https://github.com/kubernetes/kubectl/blob/31af32762da0400245ada0c342327da4b78eb182/pkg/drain/filters.go#L247

Reply:
Yeah good one, also reducing the number of calls to api-server would improve the throughput of controller. We should also remove any other status we can't evict, like Unknown?

WDYT?

Reply:
Also, good idea, will validate it tomorrow.

Reply:
I won't use `PodUnknown` as 
`// Deprecated: It isn't being set since 2015 (74da3b14b0c0f658b3bb8d2def5094686d0e9095)
`